File: 프로그래머스/1/12928. 약수의 합/약수의 합.py
def solution(n):
    if n == 1:
        return 1
    sum = 0
    check = [False] * (n+1)
    for i in range(1, (n//2)+1):
        if not check[i] and n/i == n//i: # 이전에 약수로 구해지지 않았고, 어떤 수로 나누어 떨어진다면
            if i == n/i: # 제곱근일때
                check[i] = True    
                sum += i 
                continue
            check[i] = True
            check[int(n/i)] = True
            sum += i + n/i
    return sum

File: 프로그래머스/1/12928. 약수의 합/약수의 합.java
import java.util.*;
class Solution {
    public int solution(int n) {
        // n 제곱근까지 로 나누면서 나누어떨어지는 수를 구한다, 젯수와 몫을 모두 약수에 추가
        
        Set<Integer> dividedSet = new HashSet<>();
        for(int i = 1; i <= Math.sqrt(n); i++){
            if(n % i == 0){
                dividedSet.add((int)n/i);
                dividedSet.add(i);
            }
        }
        System.out.println(dividedSet);
        int sum = 0;
        for(Integer num: dividedSet){
            sum += num;
        }
        return sum;
    }
}